api/src/main/java/org/openmrs/module/appointments/notification/AppointmentEventNotifier.java

```diff
@@ -25,4 +26,5 @@
     String getMedium();
     boolean isApplicable(final Appointment appointment);
     NotificationResult sendNotification(final Appointment appointment) throws NotificationException;
+    NotificationResult sendNotification(final Patient patient,  final String provider, final String link) throws NotificationException;
```

Comment:
Do we need another overloaded method? Why not enhance the "sendNotification(appointment)"? and create another class implementing the same? the "getMedium()" is supposed to indicate whats the medium? Also, in the sendNotification(appointment) - the appointment object has patient, provider, link etc.

Reply:
@angshu It is better to have over loaded method as we have different email body to construct and we also have BCC information as well. I think it is better to keep them separate as the original method won't get complicated

Reply:
@angshu: Shall we mark this conversation as resolved?

Reply:
The signature of the new method does not indicate any BCC. The parameters, patient and provider - can be fetched from the appointment object itself .. so if at all, only link argument needs to be passed in the second method!